include/wil/registry.h

```diff
@@ -17,16 +17,3139 @@
+        }
+
+        /**
+         * \brief Writes a REG_SZ value from a null-terminated string
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param subkey A string to append to the HKEY to attempt to write
+         *        can be nullptr if not needed
+         * \param value_name A string specifying the name of the registry value to write
+         *        can be nullptr to write to the unnamed default registry value
+         * \param data The null-terminated string value to write to the specified registry value
+         * \exception std::exception will be thrown on failure (including wil::ResultException)
+         *
+         * \remark Example usage:
+         *         wil::reg::set_value_string(key, L"subkey", L"value_name", L"test value");
+         */
+        inline void set_value_string(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, PCWSTR data)
+        {
+            return ::wil::reg::set_value(key, subkey, value_name, data);
+        }
+
+        /**
+         * \brief Writes a REG_SZ value from a null-terminated string
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param value_name A string specifying the name of the registry value to write
+         *        can be nullptr to write to the unnamed default registry value
+         * \param data The null-terminated string value to write to the specified registry value
+         * \exception std::exception will be thrown on failure (including wil::ResultException)
+         *
+         * \remark Example usage:
+         *         wil::reg::set_value_string(key, L"value_name", L"test value");
+         */
+        inline void set_value_string(HKEY key, _In_opt_ PCWSTR value_name, PCWSTR data)
+        {
+            return ::wil::reg::set_value(key, nullptr, value_name, data);
+        }
+
+        /**
+         * \brief Writes a REG_EXPAND_SZ value from a null-terminated string
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param subkey A string to append to the HKEY to attempt to write
+         *        can be nullptr if not needed
+         * \param value_name A string specifying the name of the registry value to write
+         *        can be nullptr to write to the unnamed default registry value
+         * \param data The null-terminated string value to write to the specified registry value
+         * \exception std::exception will be thrown on failure (including wil::ResultException)
+         *
+         * \remark Example usage:
+         *         wil::reg::set_value_expanded_string(key, L"subkey", L"value_name", L"%testvalue%");
+         */
+        inline void set_value_expanded_string(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, PCWSTR data) WI_NOEXCEPT

```

Comment:
Expanded string is a rare enough concept that it _might_ be worth defining it in brief for these calls.